src/example-types/metricGroups/pivot.js

```diff
@@ -303,10 +279,13 @@ let createPivotScope = (node, schema, getStats) => {
           [groupingType, 'drilldown', reversedLookupIndex],
           request
         )
-        group = filterGroupRanges({
-          drilldownKey,
-          group,
-        })
+        let filterGroupRanges = lookupTypeProp(
+          _.identity,
+          'filterGroupRanges',
```

Comment:
Filter group ranges doesn't feel generic enough to be in pivot itself - it's still makes assumptions that there are `ranges`, but the larger issues is just multibucket groupings during drilldown. It's also possible other types of groupings could still generate multiple buckets off node values that can't be simplified further.

I think a better approach would be to just do this in build group query itself by making it aware of its drilldown concept so you aren't trying to fix something down the road. If we pass drilldownKey as a prop through to buildGroupQuery, we allow each type to control its query generation directly without pivot itself having any concept of 'ranges'. Future types could have other properties - and in fact I bet fieldValuePartition has the same issue and can't be fixed with the node editing pattern in this PR

Reply:
Ok I updated the PR to pass the drilldown to each metric group and let each one decide it's own implementation in the buildGroupQuery fn.  I agree I think there are further considerations that we can eventually apply, and I'll continue to think about them while I broaden my understanding of the pivot code as a whole.